fearless/Modules/Staking/Analytics/AnalyticsValidators/AnalyticsValidatorsInteractor.swift

```diff
@@ -0,0 +1,217 @@
+            for: { accountIds },
+            engine: engine,
+            runtimeService: runtimeService,
+            chain: chain
+        )
+        operation.targetOperation.completionBlock = { [weak self] in
+            DispatchQueue.main.async {
+                do {
+                    let identitiesByAddress = try operation.targetOperation.extractNoCancellableResultData()
+                    self?.presenter.didReceive(identitiesByAddressResult: .success(identitiesByAddress))
+                } catch {
+                    self?.presenter.didReceive(identitiesByAddressResult: .failure(error))
+                }
+            }
+        }
+        operationManager.enqueue(operations: operation.allOperations, in: .transient)
+    }
+
+    private func createChainHistoryRangeOperationWrapper(
+        codingFactoryOperation: BaseOperation<RuntimeCoderFactoryProtocol>
+    ) -> CompoundOperationWrapper<ChainHistoryRange> {
+        let keyFactory = StorageKeyFactory()
+
+        let currentEraWrapper: CompoundOperationWrapper<[StorageResponse<StringScaleMapper<UInt32>>]> =
+            storageRequestFactory.queryItems(
+                engine: engine,
+                keys: { [try keyFactory.currentEra()] },
+                factory: { try codingFactoryOperation.extractNoCancellableResultData() },
+                storagePath: .currentEra
+            )
+
+        let activeEraWrapper: CompoundOperationWrapper<[StorageResponse<ActiveEraInfo>]> =
+            storageRequestFactory.queryItems(
+                engine: engine,
+                keys: { [try keyFactory.activeEra()] },
+                factory: { try codingFactoryOperation.extractNoCancellableResultData() },
+                storagePath: .activeEra
+            )
+
+        let historyDepthWrapper: CompoundOperationWrapper<[StorageResponse<StringScaleMapper<UInt32>>]> =
+            storageRequestFactory.queryItems(
+                engine: engine,
+                keys: { [try keyFactory.historyDepth()] },
+                factory: { try codingFactoryOperation.extractNoCancellableResultData() },
+                storagePath: .historyDepth
+            )
+
+        let dependecies = currentEraWrapper.allOperations + activeEraWrapper.allOperations
+            + historyDepthWrapper.allOperations
+        dependecies.forEach { $0.addDependency(codingFactoryOperation) }
+
+        let mergeOperation = ClosureOperation<ChainHistoryRange> {
+            guard
+                let currentEra = try currentEraWrapper.targetOperation.extractNoCancellableResultData()
+                .first?.value?.value,
+                let activeEra = try activeEraWrapper.targetOperation.extractNoCancellableResultData()
+                .first?.value?.index,
+                let historyDepth = try historyDepthWrapper.targetOperation.extractNoCancellableResultData()
+                .first?.value?.value
+            else {
+                throw PayoutRewardsServiceError.unknown
```

Comment:
May be leave this logic in PayoutRewardService or put to a common library to reuse in both places